be/src/vec/functions/function_interval.cpp

```diff
@@ -0,0 +1,134 @@
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <algorithm>
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "vec/columns/column.h"
+#include "vec/columns/column_const.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/columns/column_vector.h"
+#include "vec/common/assert_cast.h"
+#include "vec/core/block.h"
+#include "vec/core/column_numbers.h"
+#include "vec/core/column_with_type_and_name.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/functions/function.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris::vectorized {
+
+class FunctionInterval : public IFunction {
+public:
+    static constexpr auto name = "interval";
+    static FunctionPtr create() { return std::make_shared<FunctionInterval>(); }
+
+    String get_name() const override { return name; }
+    bool is_variadic() const override { return true; }
+    size_t get_number_of_arguments() const override { return 0; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& /*arguments*/) const override {
+        return std::make_shared<DataTypeInt32>();
+    }
+
+    Status execute_impl(FunctionContext* /*context*/, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        if (arguments.size() < 2) [[unlikely]] {
+            return Status::InternalError("interval requires at least 2 arguments");
+        }
+
+        auto res_col = ColumnInt32::create();
+        auto& res_data = res_col->get_data();
+        res_data.resize(input_rows_count);
+
+        auto compare_cwn = block.get_by_position(arguments[0]);
+        ColumnPtr compare_col_ptr = ColumnPtr {};
+        bool compare_is_const = false;
+        compare_is_const = is_column_const(*block.get_by_position(arguments[0]).column);
+        default_preprocess_parameter_columns(&compare_col_ptr, &compare_is_const, {0}, block,
+                                             arguments);
+
+        switch (compare_cwn.type->get_primitive_type()) {
+        case PrimitiveType::TYPE_TINYINT:
+            compute_interval<ColumnInt8>(block, arguments, *compare_col_ptr, compare_is_const,
```

Comment:
then, you can reference `FunctionMonthsBetween`, there's 3 possibility: data const, parameters all const, non const